stan/math/fwd/core/fvar.hpp

```diff
@@ -39,46 +39,45 @@ namespace math {
 template <typename T>
 struct fvar {
   /**
-   * The value of this variable.
-   */
-  T val_;
-
-  /**
-   * The tangent (derivative) of this variable.
+   * The type of values and tangents.
    */
-  T d_;
+  using Scalar = std::decay_t<T>;
 
   /**
-   * The type of values and tangents.
+   * The value of this variable.
    */
-  using Scalar = T;
+  Scalar val_;
 
   /**
    * Return the value of this variable.
    *
    * @return value of this variable
    */
-  T val() const { return val_; }
+  Scalar val() const { return val_; }
+
+  /**
+   * The tangent (derivative) of this variable.
+   */
+  Scalar d_;
 
   /**
    * Return the tangent (derivative) of this variable.
    *
    * @return tangent of this variable
    */
-  T d() const { return d_; }
+  Scalar tangent() const { return d_; }
```

Comment:
Probably best to leave this as `d()` for this PR, since it's a larger API change - might need wider discussion

Reply:
Yes, odd I have no idea why I named this tangent? Removed